src/Compression/ICompressionCodec.cpp

```diff
@@ -16,11 +16,15 @@ namespace ErrorCodes
 {
     extern const int CANNOT_DECOMPRESS;
     extern const int CORRUPTED_DATA;
+    extern const int CANNOT_COMPRESS;
 }
 
 
 UInt32 ICompressionCodec::compress(const char * source, UInt32 source_size, char * dest) const
 {
+    if (!source || !dest)
```

Comment:
This method assumes that source and dest are not nullptr
(similar to memcpy)

It means that the existence of this fix indicates that the root cause is somewhere else.